src/coreclr/jit/emitarm64.h

```diff
@@ -789,12 +795,17 @@ inline static bool isFloatReg(regNumber reg)
 
 inline static bool isPredicateRegister(regNumber reg)
 {
-    return (reg >= REG_PREDICATE_FIRST && reg <= REG_PREDICATE_LAST);
+    return (reg >= REG_PREDICATE_FIRST) && (reg <= REG_PREDICATE_LAST);
 }
 
 inline static bool isLowPredicateRegister(regNumber reg)
 {
-    return (reg >= REG_PREDICATE_FIRST && reg <= REG_PREDICATE_LOW_LAST);
+    return (reg >= REG_PREDICATE_FIRST) && (reg <= REG_PREDICATE_LOW_LAST);
+}
+
+inline static bool isHighPredicateRegister(regNumber reg)
+{
+    return (reg > REG_PREDICATE_LOW_LAST) && (reg <= REG_PREDICATE_LAST);
```

Comment:
probably good to introduce `REG_PREDICATE_HIGH_FIRST` and `REG_PREDICATE_HIGH_LAST` and use it here. It is more explicit that way.

Reply:
and a `static_assert_no_msg(REG_PREDICATE_HIGH_LAST == REG_PREDICATE_LAST)`.